### DISCUSSION TITLE
Need help with humming/buzzing

### DISCUSSION THREAD

**Discussion by paulhoff** [mcguirepr89/BirdNET-Pi#783] [General]
Greetings.  New to Birdnet Pi.  After running through three sound cards, I found one with decent results.  Rode AI-Micro paired with a PowerDeWise Professional Grade Lavalier Clip On Microphone, all running on a PI 4.  But I still get a huge hum/buzz on the audio.  The buzz almost overwhelms everything else when I listen to the audio feed.  On the spectrograms, it shows up as continuous lines up to about 1.5 dBFS.  

Are there any steps I can take to clean up the sound?  So far, I have tried inserting a USB extension between the PI and the Rode.  It didn't make any difference.  Nor did moving the PI as far as possible from the wall wort power supply.

Also, what level should I set the gain to?  

Thanks, everyone!

**Comment by ampsu** [comment]
It looks like there might be a issue with TRS and TRRS connections. Make sure that USB adapter and mic both use the same connection. I also had issues with buzzing and correct mic/adapter solved it. Also, I get buzzing from wall adapters but not from the power bank... So I guess there could be multiple reasons.

**Comment by matthew73210** [comment]
If you can find out the Hz of the hum, it'd give a good indication to what is happening. Try adding a choke to the USB cable, the power cable and to the mic cable.

**Comment by paulhoff** [comment]
Thanks for the suggestion.  What is a choke and where can I get one?

**Reply by blogdemoi** [reply]
I suppose this is a choke: https://en.wikipedia.org/wiki/Ferrite_bead You may find more info searching on the web with these keywords: choke usb cable

**Comment by paulhoff** [comment]
Thanks.  Searching Ferrite Choke was what did it for me.

**Comment by ampsu** [comment]
I finally found cable+adapter combination that works at home but now I have issues at my cottage. I got multiple ferrite Type-C cables, tons of different adapters but nothing seems to work. I guess the problem lies within the electrical system. I guess I must continue using power banks and switching them every day :D

**Reply by matthew73210** [reply]
Ferrites or chokes will only work agaisn't common mode interference. After adding a choke, the next step is to add X and Y caps. But that's not an easy thing to do and is only recommended if you know what you are doing. [filters](https://hardwarebee.com/WikiBee/y-capacitor/#:~:text=Y%20capacitors%20are%20used%20to,through%20the%20mains%20power%20line.)
Basically, anything metal can act either as an antenna or a radiator. We'll leave tuning out of the question for now. A simple wire can act as a monopole or dipole antenna depending on where the center point is, thus will be susceptible to electrical fields and electromagnetic waves. Any loop, and I mean any. Like traces on a board formed by circuits, multiple point connection earths and even common mode paths will form a loop antenna which are susceptible to magnetic fields and electromagnetic waves. When dealing with emf, you are never safe technically. 
What sort of circuits does your house have, what frequency is the hum, are you using a cable between mic and amp or a combined unit, are you near a substation or near hvdc powerlines, are you near a radar installation or perhaps even do you have a radioactive source near the setup?
Back to tuning, if by and chance you circuit has the resonant frequency of that of the mains. Then good luck.
The first thing to do is get the frequency of the hum and apply an anti aliasing filter. That will give a clue as to the source. I'd also do a quick gaussian filter on the signal to make sure that you aren't just picking up random noise which would point to an adc problem. Out of interest what voltage do you have between earth and neutral?

**Comment by kgun12** [comment]
I've got this same issue with that exact same mic... it disappears immediately when I plug back in my older cheaper mic, and folks on amazon seem to have the same problem if you dig deep enough in the reviews.

**Comment by nilspupils** [comment]
Hi, we had the topic with hum several times here:
https://github.com/mcguirepr89/BirdNET-Pi/discussions/39
Strongest recommendation: Get a digital Boya 
mic.

**Reply by matthew73210** [reply]
**This should be the only recommendation**. Going the analogue route you really need to have experience in electrical engineering, despite what people say. Every issue I've seen with analogue mics is a year 1 problem posed in basic electrical engineering course at uni. Anything but at the micro level in analogue and you are going to have a hard hard time with interference because everything works as an antenna. **Keep it simple, stupid --> digital mic**
